eth/polls/eth_subscribe.go

```diff
@@ -25,61 +26,79 @@ type EthPoll interface {
 }
 
 type EthSubscriptions struct {
-	polls map[string]EthPoll
-
+	polls       map[string]EthPoll
 	lastPoll    map[string]uint64
 	timestamps  map[uint64][]string
+	mutex       sync.RWMutex
 	lastPrune   uint64
 	evmAuxStore *evmaux.EvmAuxStore
 	blockStore  store.BlockStore
 }
 
 func NewEthSubscriptions(evmAuxStore *evmaux.EvmAuxStore, blockStore store.BlockStore) *EthSubscriptions {
 	p := &EthSubscriptions{
-		polls:       make(map[string]EthPoll),
-		lastPoll:    make(map[string]uint64),
-		timestamps:  make(map[uint64][]string),
+		polls:      make(map[string]EthPoll),
+		lastPoll:   make(map[string]uint64),
+		timestamps: make(map[uint64][]string),
+
 		evmAuxStore: evmAuxStore,
 		blockStore:  blockStore,
 	}
 	return p
 }
 
-func (s EthSubscriptions) Add(poll EthPoll, height uint64) string {
+func (s *EthSubscriptions) Add(poll EthPoll, height uint64) string {
 	id := utils.GetId()
-	s.polls[id] = poll
 
+	s.mutex.Lock()
```

Comment:
I think we can do
```
s.mutex.Lock()
defer s.mutex.Unlock()
```
at the beginning of the function, then instead of calling s.Remove(id), we can just inline the logic.